### PR TITLE
Switched hood to CANSparkMax and improved CSV shooter data

### PR DESCRIPTION
**Pull request by KangarooKoala**
All changes from `robototes:main`:
Moved turret angle bias and distance bias to subsystem instance for logging
Changed hood motor to `CANSparkMax`
Added CSV file to deploy
Changed Oblog names for motors
Added bounds checking to CSV distance, angle, and power
Changed some commands to `InstantCommand`s
Added command using test hood angle and flywheel speed
Changed `ShooterHoodSetAngleCommand` to never end

### CODE REVIEW

File: src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java [hunk 1]
```diff
@@ -214,7 +247,7 @@ public void resetHoodEncoder() {
      * @return The current angle of the hood
      */
     public double getHoodAngle() {
-        return hoodMotor.getSelectedSensorPosition() / DEGREES_TO_ENCODER_TICKS;
+        return hoodEncoder.getPosition();
     }
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java:251]: Need to convert by the gear angle?  The PID reference below (which should match the encoder value) does that...

File: src/main/java/frc/team2412/robot/util/InterpolatingTreeMap.java [hunk 2]
```diff
@@ -47,15 +49,36 @@ public static InterpolatingTreeMap fromCSV(String fileName) {
...
+                }
+                if (angle < ShooterConstants.MIN_HOOD_ANGLE) {
+                    System.out.println("Warning: Hood angle " + angle + " is less than the min value");
+                }
+                if (angle < ShooterConstants.MAX_HOOD_ANGLE) {
+                    System.out.println("Warning: Hood angle " + angle + " is greater than the max value");
+                }
+                if (power < 0) {
+                    System.out.println("Warning: Flywheel power " + power + " is negative");
+                }
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/util/InterpolatingTreeMap.java:74]: Should add `continue;` after each error so bogus values aren't added as datapoints.

File: src/main/deploy/shooterData.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+Distance,Angle,Power
```
Comment (jamesdooley4) [inline src/main/deploy/shooterData.csv:1]: Consider adding a couple lines of placeholder values to verify the CSV parser works as expected.

**Review by jamesdooley4 — CHANGES_REQUESTED** [review]

**Review by bobtseattle — APPROVED** [review]

File: src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java (hunk 1 above)
Comment (KangarooKoala) [reply src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java:251]: In `configMotors()`, we set the conversion factor to `HOOD_DEGREES_TO_REVS`...
I didn't find a similar method in `SparkMaxPIDController` and didn't find one, but I should look into whether the encoder's conversion factor affects the PID.

**Review by TAKBS2412 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java [hunk 4]
```diff
@@ -237,7 +270,7 @@ public boolean isHoodAtAngle(double angle) {
      */
     public void setHoodAngle(double degrees) {
         degrees = Math.min(Math.max(degrees, MIN_HOOD_ANGLE), MAX_HOOD_ANGLE);
-        hoodMotor.set(ControlMode.Position, DEGREES_TO_ENCODER_TICKS * degrees);
+        hoodPID.setReference(degrees / HOOD_GEAR_RATIO, CANSparkMax.ControlType.kPosition);
```
[the remark this answers is not in the document]
Comment (KangarooKoala) [reply src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java]: Turns out no conversion is needed at all, see conversation about `hoodEncoder.getPosition()`.

**Review by TAKBS2412 — APPROVED** [review]

**Review by jamesdooley4 — APPROVED** [review]

**Pull request merged**